### PR TITLE
support for terraform registry remote modules

### PR DESCRIPTION
**Pull request by patilpankaj212**
1. Added remote module downloader for terraform registry
2. Fix issue - failure if the remote module itself has local modules

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,8 +12,8 @@ require (
 	github.com/ghodss/yaml v1.0.0
 	github.com/gorilla/mux v1.8.0
 	github.com/hashicorp/go-cleanhttp v0.5.1
-	github.com/hashicorp/go-getter v1.5.1
-	github.com/hashicorp/go-retryablehttp v0.6.6
+	github.com/hashicorp/go-getter v1.5.2
```
Comment (dev-gaur) [inline go.mod:15]: @patilpankaj212 can you omit this go.mod change ? This caused build failure on ARM and willie's PR undid this. 
https://github.com/accurics/terrascan/pull/507/files

**Review by dev-gaur — CHANGES_REQUESTED** [review]

**Review by kanchwala-yusuf — APPROVED** [review]

**Comment by jlunaq** [comment]
Hi there,

Are the remote modules only downloadable from the official terraform registry? I need to download some modules from my private repos but it tries to read them locally.

**Comment by patilpankaj212** [comment]
Hello @jlunaq,
We remote backends - git, s3, gcs, http, terraform-registry
Could you please let us know where you have hosted your repo?
Can you please give an example of the source attribute in your module definition?

**Comment by jlunaq** [comment]
Hi @patilpankaj212 ,

Sorry for the late response, I haven't been notified.
The repo is in github, I found that terrascan is able to read the module when it is in the root directory.
So this issue is more related to the modules which are in a subdirectory, like this:

`source = "git@github.com:company/module.git//modules/base?ref=v2.0.0"`

In this case is where I'm getting the error

`<nil>: Failed to read module directory; Module directory /Users/jluna/my-project/git@github.com:company/module.git/modules/roles does not exist or cannot be read.`

**Comment by patilpankaj212** [comment]
Hello @jlunaq,
Thanks for doing the tests. Looks like a defect to me. Can you please open a issue for this case?

**Comment by jlunaq** [comment]
Sure, will do

**Comment by jlunaq** [comment]
https://github.com/accurics/terrascan/issues/600

**Comment by patilpankaj212** [comment]
Thanks @jlunaq

**Pull request merged**